.changesets/feat_garypen_2278_rhai_router_service.md

```diff
@@ -0,0 +1,43 @@
+### Provide a rhai interface to the router service ([Issue #2278](https://github.com/apollographql/router/issues/2278))
+
+Adds `Rhai` support for the `router_service`.
+
+It is now possible to interact with requests and responses at the `router_service` level from `Rhai`. The functionality is very similar to that provided for interacting with existing services, for example `supergraph_service`. For instance, you may map requests and responses as follows:
+
+```json
+fn router_service(service) {
+    const request_callback = Fn("process_request");
+    service.map_request(request_callback);
+    const response_callback = Fn("process_response");
+    service.map_response(response_callback);
+}
+
+```
+The main difference from existing services is that the router_service is dealing with HTTP Bodies, not well formatted GraphQL objects. This means that the `Request.body` or `Response.body` is not a well structured object that you may interact with, but is simply a String.
```

Comment:
can we add a sentence around the fact it s a stream of chunks? (ie we will call the coprocessor with each of the chunks?)